src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs

```diff
@@ -168,23 +168,27 @@ protected override void AddEmbeddedResourcesFromAddedModules(ArrayBuilder<Cci.Ma
 
         internal override SynthesizedAttributeData SynthesizeEmbeddedAttribute()
         {
-            if ((object)_lazyEmbeddedAttribute != null)
-            {
-                return new SynthesizedAttributeData(
-                    _lazyEmbeddedAttribute.Constructor,
-                    ImmutableArray<TypedConstant>.Empty,
-                    ImmutableArray<KeyValuePair<string, TypedConstant>>.Empty);
-            }
+            return new SynthesizedAttributeData(
+                _lazyEmbeddedAttribute.Constructors[0],
+                ImmutableArray<TypedConstant>.Empty,
```

Comment:
Thinking out loud: Is there another way of selecting the correct constructor rather than index? what happens if later on another constructor was added?
Can we filter based on signature?

Reply:
Index seems like a bad idea. In general the order of metadata is "unordered" hence depending on a specific index seems wrong. Maybe something i'm missing here that makes this safer.

Reply:
`_lazy*Attribute` are `SynthesizedEmbeddedAttribute` instances. The constructors are an immutable set created with a fixed order.